finality-aleph/src/data_io.rs

```diff
@@ -117,30 +116,30 @@ where
     block_requester: RB,
     config: DataStoreConfig,
     _phantom: PhantomData<BE>,
-    network: N,
+    messages_from_network: Arc<tokio::sync::Mutex<R>>,
     messages_for_network: UnboundedSender<Message>,
-    messages_from_network: UnboundedReceiver<(Message, Recipient)>,
 }
 
-impl<B, C, BE, RB, Message, N> DataStore<B, C, BE, RB, Message, N>
+impl<B, C, BE, RB, Message, R> DataStore<B, C, BE, RB, Message, R>
 where
     B: BlockT,
     C: crate::ClientForAleph<B, BE> + Send + Sync + 'static,
     BE: Backend<B> + 'static,
     RB: RequestBlocks<B> + 'static,
     Message:
         AlephNetworkMessage<B> + std::fmt::Debug + Send + Sync + Clone + codec::Codec + 'static,
-    N: DataNetwork<Message>,
+    R: ReceiverComponent<Message>,
 {
     /// Returns a struct to be run and a network that outputs messages filtered as appropriate
-    pub(crate) fn new(
+    pub(crate) fn new<N: ComponentNetwork<Message, R=R>>(
         client: Arc<C>,
         block_requester: RB,
         config: DataStoreConfig,
-        network: N,
+        component_network: N,
     ) -> (Self, impl DataNetwork<Message>) {
         let (messages_for_network, messages_from_data_store) = mpsc::unbounded();
-        let (messages_to_data_store, messages_from_network) = mpsc::unbounded();
+        let messages_to_data_store = component_network.sender().clone();
+        let messages_from_network = component_network.receiver();
```

Comment:
The names here are a bit confusing. I would say either use "network" to refer to the aleph network and, uh, "world"(?) to refer to the interface with the world, or use "aleph" to refer to the aleph network and "network" to refer to the network.

After I wrote this maybe the actual answer is using "aleph" and "world"... Anyway the channels would then be called (in the order they appear in the three lines above this comment):
1. `messages_for_aleph`
2. `messages_from_data_store`
3. `messages_for_world`
4. `messages_from_world`
If you pick any other of the naming schemes then change the names accordingly ofc. I just don't like using "network" in two meanings here, and implying that the messages that are going straight through the sender are "to data store".

Reply:
I like convention to use `aleph`for filtered network and just use `network` to refer to non-filtered one. Will align code to this,